### PR TITLE
Datalog shrink data

### PR DESCRIPTION
**Pull request by eerkaijun**
Contains unit tests to transform historical ES data. I assume that the historical data are stored in indices named in the format of datalog.gl_<month><year> (for eg: datalog.gl_sep20).

### CODE REVIEW

**Comment by winterstein** [comment]
Hello @eerkaijun - Could you git add `TransformRequestBuilder.java`? Thanks :)

**Comment by eerkaijun** [comment]
@winterstein `TransformRequestBuilder.java` is in the elasticsearch-java-client repository, would you like me to add it to this repository as well? If so, I should add to the datalog.server source folder?

**Comment by eerkaijun** [comment]
Currently using painless script to do the transform in order to make sure no documents are left out -- however one drawback is that the tranform job becomes very complex and timeout might occurs for large dataset. Perhaps after the ES upgrade and using the `missing_bucket` attribute instead of painless script will make things more efficient

**Comment by eerkaijun** [comment]
Note: For index with millions of documents as in the test server, the use of painless script transform will need to take hours to complete.

**Comment by eerkaijun** [comment]
Ok, so the upgraded ES version is indeed more efficient in dealing with transform. I've updated the code to perform transform for version >= 0.7.10 (warning: there will be errors if you use it on ES version < 7.10.0).

Is there a way to get the local ES version so we could write an if else statement -- so if the version >= 7.10.0, then we run setBody function, else we run setBodyWithPainless function (a less efficient way of course)

**Comment by winterstein** [comment]
> Is there a way to get the local ES version so we could write an if else statement -- so if the version >= 7.10.0, then we run setBody function, else we run setBodyWithPainless function (a less efficient way of course)

We could add a method. A call to `http://localhost:9200/` returns version.number: "7.10.0"

File: winterwell.datalog/src/java/com/winterwell/datalog/server/CompressDataLogIndexMain.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+		Log.d("compress", response);
+		
+		//after creating transform job, start it 
+		TransformRequestBuilder trb2 = esc.prepareTransformStart(jobId); 
+		trb2.setDebug(true);
+		IESResponse response2 = trb2.get().check();
+		Log.d("compress", response2);
+		System.out.println("Transforming data, please wait...");
+		
+		Utils.sleep(36000000); //allow transform job to be completed before deleting it, a safe estimate would be one hour
```
Comment (winterstein) [inline winterwell.datalog/src/java/com/winterwell/datalog/server/CompressDataLogIndexMain.java:133]: Hm... A small index should be fast, and a shame to take an hour. And maybe a monster index and/or an overloaded server could take longer than an hour.

I bet there's an API method to check the status.

Let's poll for status every 5 seconds if we can.

**Comment by eerkaijun** [comment]
Ok, this should be it. Added an ES version check and also found a _transform/stats API which will check whether the transform job is done. Apologies for not coming across it previously, must have cause you some trouble. 

Done some tests on local ES data and also data on test server. Seems to work fine.

**Comment by winterstein** [comment]
OK - let's merge this one in :)

**Pull request merged**